### PR TITLE
docker images, container and volume

### PR DESCRIPTION
**Pull request by mukutdanielroy**
Fixes #

- [ ] Does not depend on other PRs
- [x] Depends on other PR being merged first: #225 

## What kind of change does this PR introduce (check at least one)
Dockerfile was created for the todo app.
- [ ] Bugfix
- [ ] Improvement
- [x] New feature
- [ ] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?
Dockerfile was added where new images can be created.
## How to test

- [ ] Needs regression testing (add QA needed label)

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

## Reminder list

- Determine whether REST API Documentation should be updated
- Make an assessment on whether unit tests should be updated and/or added
- Ensure follow up tasks are linked to this PR

QA - if regression testing was requested:

- [ ] QA has verified

### CODE REVIEW

File: daniel-todo-app/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+#Set up workdir
+WORKDIR /usr/share/nginx/html
+#Remove all files in html
+RUN rm -rf ./*
+#Copy all from build to builder
+COPY --from=base-img app/build .
+#Run nginx
+ENTRYPOINT ["nginx", "-g", "daemon off;"]
+#copies all file to root
+EXPOSE 80
```
Comment (musfique-m) [inline daniel-todo-app/Dockerfile:26]: Please move the expose command before entrypoint.

**Review by musfique-m — CHANGES_REQUESTED** [review]
Please make separate `Dockerfile.prod` & `docker-compose.prod.yml`and move the production code there. In the `docker-compose.prod.yml` make sure it uses the `DockerFile.prod` to build image. In the `Dockerfile` & `docker-compose.yml` just keep the dev version of the docker commands. Also remove comment from prod versions.

File: daniel-todo-app/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+FROM node:alpine AS base-img
+#Builds directory in
+WORKDIR /app
+
+#Copy reqiured json packages
+COPY package.json ./
+COPY package-lock.json ./
+#installs json packages
+RUN npm install --silent
+RUN npm install react-scripts@5.0.1 -g --silent
```
[the remark this answers is not in the document]
Comment (mukutdanielroy) [reply daniel-todo-app/Dockerfile]: only package*.json kept in next commit.

**Comment by mukutdanielroy** [comment]
> Please make separate `Dockerfile.prod` & `docker-compose.prod.yml`and move the production code there. In the `docker-compose.prod.yml` make sure it uses the `DockerFile.prod` to build image. In the `Dockerfile` & `docker-compose.yml` just keep the dev version of the docker commands. Also remove comment from prod versions.

all docker files are being seperated.

File: daniel-todo-app/Dockerfile.prod [hunk 3]
```diff
@@ -0,0 +1,24 @@
+#Downloads Image
+FROM node:alpine AS base-img
+#Builds directory in
+WORKDIR /app
+
+#Copy reqiured json packages
+COPY package.json ./
```
Comment (musfique-m) [inline daniel-todo-app/Dockerfile.prod:7]: Please also use wild card here.

**Review by musfique-m — CHANGES_REQUESTED** [review]

File: daniel-todo-app/Dockerfile.prod (hunk 3 above)
Comment (mukutdanielroy) [reply daniel-todo-app/Dockerfile.prod]: wild card was added in the next commit.

**Review by musfique-m — APPROVED** [review]
Looks good.

**Pull request merged**